src/lib/containers/HasAccess.tsx

```diff
@@ -134,89 +159,83 @@ export default class HasAccess extends React.Component<
   renderIcon = (downloadType: DownloadTypeEnum) => {
     switch (downloadType) {
       case DownloadTypeEnum.NoAccess:
-        return this.renderIconHelper(faMinusCircle, 'SRC-warning-color')
+        return this.renderIconHelper(faMinusCircle, 'SRC-danger-color')
       case DownloadTypeEnum.ExternalFileHandle:
         return this.renderIconHelper(faLink, 'SRC-warning-color')
       case DownloadTypeEnum.CloudFileHandle:
         return this.renderIconHelper(faLink, 'SRC-warning-color')
-      case DownloadTypeEnum.IsOpenNoRestrictions:
-        return this.renderIconHelper(faUnlockAlt, 'SRC-success-color')
+      case DownloadTypeEnum.HasUnmetAccessRestrictions:
+        return this.renderIconHelper(faMinusCircle, 'SRC-warning-color')
       case DownloadTypeEnum.TooLargeFile:
         return this.renderIconHelper(faDatabase, 'SRC-danger-color')
+      case DownloadTypeEnum.IsOpenNoUnmetAccessRestrictions:
+        return this.renderIconHelper(faUnlockAlt, 'SRC-success-color')
       default:
         throw Error('downloadTypeEnum passed incorrectly')
     }
   }
 
-  // Utility to declare what type of download this is
+  // Get type of download
   getDownloadType = () => {
-    const { fileHandle } = this.props
+    // check if they have unmet access requirements
     const { restrictionInformation } = this.state
-    const isOpen =
-      fileHandle ||
-      (restrictionInformation &&
-        RestrictionLevel.OPEN === restrictionInformation.restrictionLevel)
-    const isTooLarge = fileHandle && fileHandle.contentSize >= GIGABYTE_SIZE
-    if (isTooLarge) {
-      return DownloadTypeEnum.TooLargeFile
-    } else if (isOpen) {
-      const concreteType = fileHandle ? fileHandle.concreteType : ''
-      const isGoogleCloudFileHandle =
-        concreteType === GoogleCloudFileHandleEnum.GoogleCloudFileHandle
-      if (isGoogleCloudFileHandle) {
-        return DownloadTypeEnum.CloudFileHandle
+    if (restrictionInformation) {
+      const { hasUnmetAccessRequirement } = restrictionInformation
+      if (hasUnmetAccessRequirement) {
+        return DownloadTypeEnum.HasUnmetAccessRestrictions
       } else {
-        const isExternalFileHandle = Object.values(
-          ExternalFileHandleConcreteTypeEnum,
-          // @ts-ignore
-        ).includes(concreteType)
-        if (isExternalFileHandle) {
-          return DownloadTypeEnum.ExternalFileHandle
-        }
-        return DownloadTypeEnum.IsOpenNoRestrictions
+        return DownloadTypeEnum.IsOpenNoUnmetAccessRestrictions
       }
-    } else {
-      return DownloadTypeEnum.NoAccess
     }
+    // if file handle is present show more detailed download information
+    const { fileHandle } = this.props
+    if (fileHandle) {
+      return getFileHandleType(fileHandle)
+    }
+    // otherwise no access to this file
+    return DownloadTypeEnum.NoAccess
```

Comment:
error here instead

Reply:
handle init (before restrictionInfo is available)